geetools/Image.py

```diff
@@ -1315,11 +1317,27 @@ def distance(self, other: ee.image) -> ee.Image:
 
         return ee.Image(distance)
 
-    def maskCover(self) -> ee.Image:
```

Comment:
Ok, so you're saying that instead of having a `region` param we should just use the image's geometry and the user should use `.clip` to set it? :thinking: not sure about that, I usually avoid using the `.clip` method. In my imagination the process of clipping the image is different than using a geometry to "select" a bunch of pixels for further processing, but maybe I am wrong since I cannot be sure how it works on the server side (maybe we can ask). I just tried to mimic the `reduceRegion` method, in which there is a `geometry` param that defaults to the image's geometry (I didn't use the same name, I used `region`, that's arguable). What I see now is that I didn't set a default value for `region` which could be the image's geometry as in `reduceRegion`.

Reply:
for the one you buld that's perfect and the naming convention makes it super clear what it's doing so kudos. For this specific one that works fine to `.clip` first. Nick already confirmed it's changing the image footprint. It could simply called maskCoverRegion using it's own geometry and push it as a metadata of the image.